### PR TITLE
[polyhook2] Update to 2020-05-25

### PR DESCRIPTION
**Pull request by mcmtroffaes**
- Update to latest PolyHook_2 revision.

- Add support for zydis and inlinentd as features (enabled by default, as per upstream default).

- Support for capstone, exception, detours, and pe is now configurable via features (all enabled by default, as per upstream default).

- Remove cmake build patch. Relevant parts have been upstreamed.

- For now, remove tool feature for building tests, as upstream does not support building tests along with building the library. (The original port carried a very large patch to the cmake build script to make this work, which is likely to keep breaking with upstream changes.)

- Add support for cmake config so library can be used simply with "find_package(PolyHook_2 CONFIG REQUIRED)".

**Describe the pull request**

- What does your PR fix? See above.

- Which triplets are supported/not supported? Have you updated the CI baseline? All standard triplets tested and supported.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes.

cc upstream @stevemk14ebr

### CODE REVIEW

**Comment by stevemk14ebr** [comment]
Awesome! I am fine with these settings

File: ports/polyhook2/CONTROL [hunk 1]
```diff
@@ -1,8 +1,29 @@
 Source: polyhook2
-Version: 2020-02-17
+Version: 2020-05-25
 Homepage: https://github.com/stevemk14ebr/PolyHook_2_0
 Description: C++17, x86/x64 Hooking Libary v2.0
```
Comment (NancyLi1013) [inline ports/polyhook2/CONTROL:4]:
```suggestion
Description: C++17, x86/x64 Hooking Libary v2.0
Supports: !(arm|uwp|linux|osx)
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by mcmtroffaes** [comment]
Many thanks for the timely review. If anything else is needed, let me know.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Need to test features manually. I will update the results once finished.

Currently, there is no need to do anything.

**Comment by NancyLi1013** [comment]
All features have passed with the following triplets:

- x86-windows
- x64-windows
- x64-windows-static

**Comment by mcmtroffaes** [comment]
I'm glad everything passed review. Anything else needed?

**Comment by NancyLi1013** [comment]
Nothing needs to be done, just waiting for the merged.

**Pull request merged**